File: src/dlp/ast/sentences/IfElse.ts

import {Sentence} from "./Sentence";
import {Expression} from "../expressions/Expression";
import {ConditionalSentence} from "./ConditionalSentence";

export abstract class IfElse extends ConditionalSentence implements Sentence{

    elseBody: Sentence;

    constructor(condition: Expression, body: Sentence, elseBody: Sentence, line: number, column: number) {
        super(condition, body, line,column);
        this.elseBody = elseBody;
    }
}

File: src/ast/sentences/IfElse.java
package ast.sentences;

import ast.expressions.Expression;

public class IfElse extends ConditionalSentence {

    private Sentence elseBody;

    public IfElse(Expression condition, Sentence body, Sentence elseBody, int line, int column) {
        super(condition, body, line, column);
        this.elseBody = elseBody;
    }

    public Sentence getElseBody() {
        return elseBody;
    }
}
